packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -830,7 +830,7 @@ class RenderShaderMask extends RenderProxyBox {
   void paint(PaintingContext context, Offset offset) {
     if (child != null) {
       assert(needsCompositing);
-      final Rect rect = Offset.zero & size;
+      final Rect rect = offset & size;
```

Comment:
I'm not sure this is completely correct.  We pass the offset to pushShaderMask as well, which is redundant with the rect that we pass now.  I'd look at how the other functions (e.g., clip) handle this case.  I thought we always passed the rects before applying the offset.

I guess the tricky issue here is the shader callback?